### PR TITLE
tests: pick up k8s test logs, other enhancements

### PR DESCRIPTION
**Pull request by ianvernon**
* tests/k8s: copy contents of /var/lib/cilium, /var/run/cilium, as well as logs from kube-apiserver, kube-controller-manager, kubelet into tar.gz that is archived after each build.
* tests: shut down VMs even if gathering of logs / archiving of artifacts fails.

Signed-off by: Ian Vernon <ian@covalent.io>

Fixes: #1047 
Fixes: #1040 
Fixes: #1004

### CODE REVIEW

**Comment by ianvernon** [comment]
Example of successful build with gathering of logs: https://jenkins.cilium.io/job/cilium/job/cilium/job/PR-1045/26/

Example of failure in archiving logs but VMs still getting destroyed: https://jenkins.cilium.io/job/cilium/job/cilium/job/PR-1045/24/console

File: tests/k8s/k8s-gsg-test.sh [hunk 1]
```diff
@@ -12,14 +14,29 @@ function cleanup {
...
+trap finish_test exit
 
 echo "KUBECONFIG: $KUBECONFIG"
 
 cleanup
 
+echo "---- Waiting for cluster to get into a good state ----"
 until [ "$(kubectl get cs | grep -v "STATUS" | grep -c "Healthy")" -eq "3" ]; do 
-	echo "---- Waiting for cluster to get into a good state ----"
+  continue
```
Comment (aanm) [inline tests/k8s/k8s-gsg-test.sh]: I think all `continue` should be replaced with an `echo -n "."` so that we know things are moving.

Comment (ianvernon) [reply tests/k8s/k8s-gsg-test.sh]: This prints out a '.' on a single line, which means the logs are still verbose. Is this desired? #1004  was filed to reduce the amount of noise in the test logs, and this means the same # of lines will be output by the loop.

Comment (aanm) [reply tests/k8s/k8s-gsg-test.sh]: the `-n` omits the new line, @tgraf already made this changed but jenkins picks each `.` as a new line so we have a trailing amount of `.`
```
.
.
.
.
.
.
```

Comment (ianvernon) [reply tests/k8s/k8s-gsg-test.sh]: OK, I will rebase and leave as is in that case.

**Comment by tgraf** [comment]
Looks good. There is some overlap with #1078.

File: Jenkinsfile [hunk 2]
```diff
@@ -22,8 +22,10 @@ pipeline {
     }
     post {
         always {
-            sh './tests/copy_files'
-            archiveArtifacts "cilium-files.tar.gz"
+            sh './tests/copy_files || true'
+            archiveArtifacts artifacts: "cilium-files-runtime.tar.gz", allowEmptyArchive: true
+            sh './tests/k8s/copy_files || true'
```
Comment (aanm) [inline Jenkinsfile:27]: I think @tgraf solve this part by writing `exit 0` in the `copy_files` but both me and him saw a PR timing out and jenkins assumed that all tests passed successfully and we weren't sure if it was due `exit 0`. I'm fine keeping this `|| true`

**Review by tgraf — APPROVED** [review]

**Pull request merged**